econml/sklearn_extensions/model_selection.py

```diff
@@ -190,3 +201,183 @@ def split(self, X, y, sample_weight=None):
             Weights associated with the training data.
         """
         return _split_weighted_sample(self, X, y, sample_weight, is_stratified=True)
+
+
+def cross_val_predict(estimator, X, y=None, *, groups=None, cv=None,
+                      n_jobs=None, verbose=0, fit_params=None,
```

Comment:
This looks extremely similar to sklearn's version - what's different?  Also, be sure to add a comment giving attribution to sklearn so that it's obvious we're not claiming this is entirely our work.

Reply:
It is a copy-paste, with the only change in passing the "safe" keyword arg, that allows for safe=False, cloning of the models for each fold. Otherwise it wont be able to accept models that are not sklearn models.